packages/api/src/controllers/webhook.ts

```diff
@@ -237,6 +234,39 @@ app.patch(
   }
 );
 
+app.post(
+  "/:id/status",
+  authorizer({ anyAdmin: true }),
+  validatePost("webhook-status-payload"),
+  async (req, res) => {
+    const { id } = req.params;
+    const webhook = await db.webhook.get(id);
+    if (!webhook) {
+      return res.status(404).json({ errors: ["not found"] });
+    }
+
+    const doc = req.body.status;
+    const webhookResponse = req.body.response;
+
+    if (!doc) {
+      return res.status(422).json({ errors: ["missing status in payload"] });
+    }
+
+    await db.webhook.update(req.params.id, {
+      ...webhook,
+      status: doc,
+    });
+
+    if (webhookResponse) {
+      webhookResponse.id = uuid();
+      let w: WithID<WebhookResponse> = webhookResponse;
+      await db.webhookResponse.create(w);
```

Comment:
typescript rescue
```suggestion
      await db.webhookResponse.create({
         ...webhookResponse,
         id: uuid()
      });
```

and if it still complains on the typing, can change last line to 
```ts
} as WithID<WebhookResponse>);
```